activitylog/service.go

```diff
@@ -0,0 +1,84 @@
+// Copyright (c) Abstract Machines
+// SPDX-License-Identifier: Apache-2.0
+
+package activitylog
+
+import (
+	"context"
+
+	"github.com/absmach/magistrala"
+	"github.com/absmach/magistrala/auth"
+	"github.com/absmach/magistrala/pkg/errors"
+	svcerr "github.com/absmach/magistrala/pkg/errors/service"
+)
+
+type service struct {
+	auth       magistrala.AuthServiceClient
+	repository Repository
+}
+
+func NewService(repository Repository, authClient magistrala.AuthServiceClient) Service {
+	return &service{
+		auth:       authClient,
+		repository: repository,
+	}
+}
+
+func (svc *service) Save(ctx context.Context, activity Activity) error {
+	return svc.repository.Save(ctx, activity)
+}
+
+func (svc *service) RetrieveAll(ctx context.Context, token string, page Page) (ActivitiesPage, error) {
+	userID, domainID, err := svc.identify(ctx, token)
+	if err != nil {
+		return ActivitiesPage{}, err
+	}
+	if page.EntityID != "" {
+		if err := svc.authorize(ctx, userID, domainID, auth.ViewPermission, page.EntityID, page.EntityType.AuthString()); err != nil {
+			return ActivitiesPage{}, err
+		}
+
+		return svc.repository.RetrieveAll(ctx, page)
+	}
+
+	if err := svc.authorize(ctx, userID, domainID, auth.AdminPermission, auth.PlatformType, auth.MagistralaObject); err != nil {
+		page.WithAttributes = false
+		page.WithMetadata = false
+	}
+
+	return svc.repository.RetrieveAll(ctx, page)
+}
```

Comment:
@rodneyosodo Do you think we should all activities if user is not super admin ? 
If yes, why we should all activites for non super admin, How it will be useful from them .

Reply:
What is being retrieved doesn't contain sensitive information since with omit attributes and metadata we only retrieve the operation and when it occurred

Reply:
I'm just thinking, How it will be usefull for user. It might usefull for case like how in `ecommerce` site they will show popup with hidden sensitive infromation of recently happened purchases.
But it might be useful much in our case. 

So can we make it configurable in env with by default `false`? 

So if application want to show all events infromation, they can enable it by env variable